src/analyses/libraryFunctions.ml

```diff
@@ -30,6 +30,10 @@ let posix_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
 let pthread_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
     ("pthread_create", special [__ "thread" [w]; drop "attr" [r]; __ "start_routine" [s]; __ "arg" []] @@ fun thread start_routine arg -> ThreadCreate { thread; start_routine; arg }); (* For precision purposes arg is not considered accessed here. Instead all accesses (if any) come from actually analyzing start_routine. *)
     ("pthread_exit", special [__ "retval" []] @@ fun retval -> ThreadExit { ret_val = retval }); (* Doesn't dereference the void* itself, but just passes to pthread_join. *)
+    ("pthread_cond_signal", special [__ "cond" []] @@ fun cond -> Signal cond);
+    ("pthread_cond_broadcast", special [__ "cond" []] @@ fun cond -> Broadcast cond);
+    ("pthread_cond_wait", special [__ "cond" []; __ "mutex" []] @@ fun cond mutex -> Wait {cond; mutex});
+    ("pthread_cond_timedwait", special [__ "cond" []; __ "mutex" []; __ "abstime" []] @@ fun cond mutex abstime -> TimedWait {cond; mutex; abstime});
```

Comment:
@sim642 I would appreciate your feedback on the switching to the new library mechanism here.

Reply:
The last line should have `"abstime" [r]` because that function reads the time struct through the pointer, other than that these look right!

This reminds me, sometime ago I started doing some cleanup, including these cond waits: https://github.com/goblint/analyzer/commit/6fb277fc75dd00a3e51df045f5a9e2594838d708#diff-f11dd379df0d8dcedf47eec4569e585bdf20a8671f9194a6cdfd69e65ce0c4e5R33-R34. I should probably revive that.